source/PluginDev/Assets/GooglePlayGames/Platforms/Android/AndroidTokenClient.cs

```diff
@@ -132,6 +133,70 @@ public void FetchTokens(bool silent, Action<int> callback)
             PlayGamesHelperObject.RunOnGameThread(() => DoFetchToken(silent, callback));
         }
 
+        public void RequestPermissions(Action<RequestPermissionUIStatus> callback, string[] scopes)
+        {
+            using (var bridgeClass = new AndroidJavaClass(HelperFragmentClass))
+            using (var currentActivity = AndroidHelperFragment.GetActivity())
+            using (var task =
+                bridgeClass.CallStatic<AndroidJavaObject>("showRequestPermissionsUi", currentActivity,
+                    oauthScopes.Union(scopes).ToArray()))
+            {
+                AndroidTaskUtils.AddOnSuccessListener<AndroidJavaObject>(task, /* disposeResult= */ false,
+                    accountWithNewScopes =>
+                    {
+                        if (accountWithNewScopes == null)
+                        {
+                            callback(RequestPermissionUIStatus.InternalError);
+                            return;
+                        }
+
+                        account = accountWithNewScopes;
+                        email = account.Call<string>("getEmail");
+                        idToken = account.Call<string>("getIdToken");
+                        authCode = account.Call<string>("getServerAuthCode");
+                        oauthScopes = oauthScopes.Union(scopes).ToList();
+                        callback(RequestPermissionUIStatus.Success);
+                    });
+
+                AndroidTaskUtils.AddOnFailureListener(task, e =>
+                {
+                    var failCode = ToRequestPermissionUiStatus(e.Call<int>("getStatusCode"));
+                    callback(failCode);
+                });
+            }
+        }
+
+        private static RequestPermissionUIStatus ToRequestPermissionUiStatus(int code)
+        {
+            Dictionary<int, RequestPermissionUIStatus> dictionary = new Dictionary<int, RequestPermissionUIStatus>()
+            {
+                {-12, RequestPermissionUIStatus.SignInCurrentlyInProgress},
```

Comment:
We should add a comments to status code names to simplify mapping understanding

Reply:
Each on's Java equivalent is added.